sdk/clientcore/annotation-processor-test/src/test/java/io/clientcore/annotation/processor/test/TestInterfaceServiceClientGenerationTests.java

```diff
@@ -1320,12 +1292,11 @@ public void service20GetResponseBody() {
     }
 
     @Test
-    @Disabled("https://github.com/Azure/azure-sdk-for-java/issues/43728")
-    public void unexpectedHTTPOK() {
-        HttpResponseException e = assertThrows(HttpResponseException.class,
+    public void unexpectedHTTPOKDeserializeError() {
+        CoreException e = assertThrows(CoreException.class,
             () -> createService(TestInterfaceClientImpl.TestInterfaceClientService.class).getBytes(getRequestUri()));
 
-        assertEquals("Status code 200, (1024-byte body)", e.getMessage());
+        assertTrue(e.getMessage().startsWith("Unexpected character"));
```

Comment:
This test logic to me feels off. The message here seems to be a deserialization error when handling the exception type.